covidthailand.py

```diff
@@ -2597,794 +2601,1157 @@ def topprov(df, metricfunc, valuefunc=None, name="Top 5 Provinces", num=5, other
-        y=rearrange(cols, *FIRST_AREAS),
-        kind="area",
-        figsize=[20, 10],
-        colormap=AREA_COLOURS,
-        title="PCR Positive Test Results by Health District\n"
-        f"Updated: {TODAY().date()}\n"
-        "djay.github.io/covidthailand"
-    )
-    ax.legend(AREA_LEGEND_UNKNOWN)
-    #ax.subtitle("Excludes proactive & private tests")
-    plt.tight_layout()
-    plt.savefig("outputs/pos_area.png")
+    AREA_COLOURS='tab20'
 
-
-    cols = [f"Tests Daily {area}" for area in range(1, 14)]
-    test_cols = [f"Tests Area {area}" for area in range(1, 14)]
+    cols = [f'Tests Area {area}' for area in range(1, 14)]
+    # fig, ax = plt.subplots()
+    # df.plot(
+    #     ax=ax,
+    #     use_index=True,
+    #     y=rearrange(cols, *FIRST_AREAS),
+    #     kind='area',
+    #     figsize=[20, 10],
+    #     colormap=AREA_COLOURS,
+    #     title='PCR Tests by Health District\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend(AREA_LEGEND_UNKNOWN)
+    # ax.subtitle('Excludes proactive & private tests')
+    # plt.tight_layout()
+    # plt.savefig('outputs/tests_area.png')
+    cols = rearrange(cols, *FIRST_AREAS),
+    plot_area(df=df, png_prefix='tests_area', cols_subset=cols[0],
+              title='PCR Tests by Health District (excludes proactive & private tests)', legends=AREA_LEGEND_UNKNOWN,
+              kind='area', stacked=True, percent_fig=False, ma=False, cmap='tab20')
+
+
+    cols = [f'Pos Area {area}' for area in range(1, 14)]
+    # fig, ax = plt.subplots()
+    # df.plot(
+    #     ax=ax,
+    #     #use_index=True,
+    #     y=rearrange(cols, *FIRST_AREAS),
+    #     kind='area',
+    #     figsize=[20, 10],
+    #     colormap=AREA_COLOURS,
+    #     title='PCR Positive Test Results by Health District\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend(AREA_LEGEND_UNKNOWN)
+    # #ax.subtitle('Excludes proactive & private tests')
+    # plt.tight_layout()
+    # plt.savefig('outputs/pos_area.png')
+    cols = rearrange(cols, *FIRST_AREAS)
+    plot_area(df=df, png_prefix='pos_area', cols_subset=cols,
+              title='PCR Positive Test Results by Health District (excludes proactive & private tests)',
+              legends=AREA_LEGEND_UNKNOWN,
+              kind='area', stacked=True, percent_fig=False, ma=False, cmap='tab20')
+
+
+    cols = [f'Tests Daily {area}' for area in range(1, 14)]
+    test_cols = [f'Tests Area {area}' for area in range(1, 14)]
     for area in range(1, 14):
-        df[f"Tests Daily {area}"] = (
-            df[f"Tests Area {area}"]
+        df[f'Tests Daily {area}'] = (
+            df[f'Tests Area {area}']
             / df[test_cols].sum(axis=1)
-            * df["Tests (MA)"]
+            * df['Tests (MA)']
         )
     df['Tests Daily 14'] = df['Tests (MA)'].sub(df[cols].sum(axis=1), fill_value=0).clip(lower=0)
```

Comment:
We shouldn't use 14 anymore. This is just working out unknowns. We should use the unknown part of area_plot to do the same?